File: app/src/main/java/com/mabi87/videocropviewsample/MainActivity.java
package com.mabi87.videocropviewsample;

import android.content.Intent;
import android.database.Cursor;
import android.media.MediaPlayer;
import android.net.Uri;
import android.os.Bundle;
import android.provider.MediaStore;
import android.support.v7.app.ActionBarActivity;
import android.view.View;

import com.mabi87.videocropview.VideoCropView;


public class MainActivity extends ActionBarActivity {

    // Layout Components
    private VideoCropView mVideoCropView;

    // Attributes
    private String originalPath;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mVideoCropView = (VideoCropView) findViewById(R.id.cropVideoView);
        mVideoCropView.setOnPreparedListener(new MediaPlayer.OnPreparedListener() {

            @Override
            public void onPrepared(MediaPlayer mp) {
                mVideoCropView.start();
            }
        });
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        if (requestCode == 1000 && resultCode == RESULT_OK) {
            Uri selectedVideoUri = data.getData();

            originalPath = getRealPathFromURI(selectedVideoUri);

            mVideoCropView.setVideoURI(selectedVideoUri);
            mVideoCropView.seekTo(1);
        }
    }

    public void onButtonLoadClick(View v) {
        Intent lIntent = new Intent(Intent.ACTION_PICK);
        lIntent.setType("video/*");
        lIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
        startActivityForResult(lIntent, 1000);
    }

    public String getRealPathFromURI(Uri contentUri) {
        Cursor cursor = null;
        try {
            String[] proj = { MediaStore.Images.Media.DATA };
            cursor = getApplicationContext().getContentResolver().query(contentUri, proj, null, null, null);
            int column_index = cursor.getColumnIndexOrThrow(MediaStore.Images.Media.DATA);
            cursor.moveToFirst();
            return cursor.getString(column_index);
        } finally {
            if (cursor != null) {
                cursor.close();
            }
        }
    }

}

File: app/src/main/java/com/crust87/videocropviewsample/MainActivity.kt
package com.crust87.videocropviewsample

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

const val REQUEST_CODE_LOAD_VIDEO = 1000

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        videoCropView.setOnPreparedListener { videoCropView.start() }

        button43.setOnClickListener {
            videoCropView.setRatio(4f, 3f)
        }

        button11.setOnClickListener {
            videoCropView.setRatio(1f, 1f)
        }

        button34.setOnClickListener {
            videoCropView.setRatio(3f, 4f)
        }

        buttonOriginal.setOnClickListener {
            videoCropView.setOriginalRatio()
        }

        buttonLoad.setOnClickListener {
            val intent = Intent(Intent.ACTION_PICK).apply {
                type = "video/*"
                addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
            }

            startActivityForResult(intent, REQUEST_CODE_LOAD_VIDEO)
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        if (requestCode == REQUEST_CODE_LOAD_VIDEO && resultCode == Activity.RESULT_OK) {
            data?.data?.let {
                videoCropView.setVideoURI(it)
                videoCropView.seekTo(1)
            }
        } else {
            super.onActivityResult(requestCode, resultCode, data)
        }
    }
}